Enable <textarea> in vertical writing mode

nt1m opened the pull request:
#### 99cf25ddd405c557255c9ae6e865e7b922f8b145
<pre>
Enable &lt;textarea&gt; in vertical writing mode
<a href="https://bugs.webkit.org/show_bug.cgi?id=248260">https://bugs.webkit.org/show_bug.cgi?id=248260</a>
rdar://102620075

Reviewed by Aditya Keerthi.

It already works properly in vertical writing mode for the most part. Enable it behind the VerticalFormControlsEnabled setting.

Upstream revision for WPT: <a href="https://github.com/web-platform-tests/wpt/commit/c758b0d0864c879ffe505c66ba6cf9345db0c665">https://github.com/web-platform-tests/wpt/commit/c758b0d0864c879ffe505c66ba6cf9345db0c665</a>

* LayoutTests/imported/w3c/web-platform-tests/css/css-writing-modes/forms/manual/support/form-controls-slr.png: Renamed from LayoutTests/imported/w3c/web-platform-tests/css/css-writing-modes/support/form-controls-slr.png.
* LayoutTests/imported/w3c/web-platform-tests/css/css-writing-modes/forms/manual/support/form-controls-srl.png: Renamed from LayoutTests/imported/w3c/web-platform-tests/css/css-writing-modes/support/form-controls-srl.png.
* LayoutTests/imported/w3c/web-platform-tests/css/css-writing-modes/forms/manual/support/form-controls-vlr.png: Renamed from LayoutTests/imported/w3c/web-platform-tests/css/css-writing-modes/support/form-controls-vlr.png.
* LayoutTests/imported/w3c/web-platform-tests/css/css-writing-modes/forms/manual/support/vertical-form.png: Renamed from LayoutTests/imported/w3c/web-platform-tests/css/css-writing-modes/support/vertical-form.png.
* LayoutTests/imported/w3c/web-platform-tests/css/css-writing-modes/forms/manual/support/w3c-import.log: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-writing-modes/forms/textarea-appearance-native-horizontal.optional-expected-mismatch.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-writing-modes/forms/textarea-appearance-native-horizontal.optional.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-writing-modes/forms/textarea-appearance-native-vlr.optional-expected-mismatch.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-writing-modes/forms/textarea-appearance-native-vlr.optional.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-writing-modes/forms/textarea-appearance-native-vrl.optional-expected-mismatch.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-writing-modes/forms/textarea-appearance-native-vrl.optional.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-writing-modes/forms/textarea-appearance-none-horizontal.optional-expected-mismatch.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-writing-modes/forms/textarea-appearance-none-horizontal.optional.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-writing-modes/forms/textarea-appearance-none-vlr.optional-expected-mismatch.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-writing-modes/forms/textarea-appearance-none-vlr.optional.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-writing-modes/forms/textarea-appearance-none-vrl.optional-expected-mismatch.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-writing-modes/forms/textarea-appearance-none-vrl.optional.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-writing-modes/forms/textarea-rows-cols-sizing-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-writing-modes/forms/textarea-rows-cols-sizing.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-writing-modes/forms/w3c-import.log:
* LayoutTests/imported/w3c/web-platform-tests/css/css-writing-modes/support/w3c-import.log:
* Source/WebCore/css/horizontalFormControls.css:
(@namespace &quot;<a href="http://www.w3.org/1999/xhtml&quot">http://www.w3.org/1999/xhtml&quot</a>;;):
* Source/WebCore/css/html.css:
(input:not([type=&quot;color&quot;]), select, button, meter, progress):
(input:not([type=&quot;color&quot;]), textarea, select, button, meter, progress): Deleted.

Canonical link: <a href="https://commits.webkit.org/256977@main">https://commits.webkit.org/256977@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/92f96cd4ed1f8ac128b0ef17e3dbdc86db4d8424

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/97427 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/6686 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/30591 "Built successfully") | [  ~~🛠 wpe~~](https://ews-build.webkit.org/#/builders/8/builds/106941 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [❌ 🛠 🧪 win](https://ews-build.webkit.org/#/builders/10/builds/167207 "Failed to checkout and rebase branch from PR 6760") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/101390 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/7008 "Built successfully") | [✅ 🛠 mac-debug](https://ews-build.webkit.org/#/builders/71/builds/35435 "Built successfully") | [  ~~🛠 gtk~~](https://ews-build.webkit.org/#/builders/36/builds/89832 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/103615 "Built successfully") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/103087 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/78/builds/5260 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/84057 "Built successfully") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/36/builds/89832 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/9/builds/87126 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/3/builds/88926 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/36/builds/89832 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🛠 tv~~](https://ews-build.webkit.org/#/builders/81/builds/695 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/73/builds/20392 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [  ~~🛠 tv-sim~~](https://ews-build.webkit.org/#/builders/82/builds/680 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/70/builds/21864 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/80/builds/5488 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/60/builds/44343 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/75/builds/2368 "Built successfully and passed tests") | [  ~~🛠 watch-sim~~](https://ews-build.webkit.org/#/builders/79/builds/1928 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk2-stress~~](https://ews-build.webkit.org/#/builders/62/builds/41217 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
<!--EWS-Status-Bubble-End-->

pxlcoder reviewed (approved):
> for most part

for the most part

nt1m committed:
Enable <textarea> in vertical writing mode
https://bugs.webkit.org/show_bug.cgi?id=248260
rdar://102620075

Reviewed by Aditya Keerthi.

It already works properly in vertical writing mode for the most part. Enable it behind the VerticalFormControlsEnabled setting.

Upstream revision for WPT: https://github.com/web-platform-tests/wpt/commit/c758b0d0864c879ffe505c66ba6cf9345db0c665

* LayoutTests/imported/w3c/web-platform-tests/css/css-writing-modes/forms/manual/support/form-controls-slr.png: Renamed from LayoutTests/imported/w3c/web-platform-tests/css/css-writing-modes/support/form-controls-slr.png.
* LayoutTests/imported/w3c/web-platform-tests/css/css-writing-modes/forms/manual/support/form-controls-srl.png: Renamed from LayoutTests/imported/w3c/web-platform-tests/css/css-writing-modes/support/form-controls-srl.png.
* LayoutTests/imported/w3c/web-platform-tests/css/css-writing-modes/forms/manual/support/form-controls-vlr.png: Renamed from LayoutTests/imported/w3c/web-platform-tests/css/css-writing-modes/support/form-controls-vlr.png.
* LayoutTests/imported/w3c/web-platform-tests/css/css-writing-modes/forms/manual/support/vertical-form.png: Renamed from LayoutTests/imported/w3c/web-platform-tests/css/css-writing-modes/support/vertical-form.png.
* LayoutTests/imported/w3c/web-platform-tests/css/css-writing-modes/forms/manual/support/w3c-import.log: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-writing-modes/forms/textarea-appearance-native-horizontal.optional-expected-mismatch.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-writing-modes/forms/textarea-appearance-native-horizontal.optional.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-writing-modes/forms/textarea-appearance-native-vlr.optional-expected-mismatch.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-writing-modes/forms/textarea-appearance-native-vlr.optional.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-writing-modes/forms/textarea-appearance-native-vrl.optional-expected-mismatch.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-writing-modes/forms/textarea-appearance-native-vrl.optional.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-writing-modes/forms/textarea-appearance-none-horizontal.optional-expected-mismatch.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-writing-modes/forms/textarea-appearance-none-horizontal.optional.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-writing-modes/forms/textarea-appearance-none-vlr.optional-expected-mismatch.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-writing-modes/forms/textarea-appearance-none-vlr.optional.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-writing-modes/forms/textarea-appearance-none-vrl.optional-expected-mismatch.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-writing-modes/forms/textarea-appearance-none-vrl.optional.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-writing-modes/forms/textarea-rows-cols-sizing-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-writing-modes/forms/textarea-rows-cols-sizing.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-writing-modes/forms/w3c-import.log:
* LayoutTests/imported/w3c/web-platform-tests/css/css-writing-modes/support/w3c-import.log:
* Source/WebCore/css/horizontalFormControls.css:
(@namespace "http://www.w3.org/1999/xhtml";):
* Source/WebCore/css/html.css:
(input:not([type="color"]), select, button, meter, progress):
(input:not([type="color"]), textarea, select, button, meter, progress): Deleted.

Canonical link: https://commits.webkit.org/256977@main
